### DISCUSSION TITLE
Why is Copilot free?

### DISCUSSION THREAD

**Discussion by jesi-rgb** [yetone/avante.nvim#931] [Q&A] [answered]
AFAIK, copilot requires a subscription of, at least, $10/mo. You suggest in the README that we can use a 'free' provider like copilot for frequent tasks. How come?

I ask this since I have a subscription for copilot and wondered if I could use the same api keys for avante, given that copilot gives access to gpt-4o, o1-mini and o1-preview, and even claude.

thanks beforehand!

**Comment by pidgeon777** [comment]
GitHub Copilot is not free *unless you are a verified student, teacher, or maintainer of a popular open source project on GitHub*, but they offer a free one-month trial period. Then you'll need to pay 10$ / month for the "Individual" subscription:

https://docs.github.com/en/copilot/about-github-copilot/subscription-plans-for-github-copilot

**Reply by josealvaradoo** [reply, marked as answer]
Copilot is now free for everyone (with limits).

**Reply by pidgeon777** [reply]
Yes, since the 18th of December.

**Comment by Daydreamer-riri** [comment]
Off topic, when I use Copilot as a provider, "suggestions" is very slow to generate, do you have a similar experience?

**Reply by pidgeon777** [reply]
I think the API is faster than using Copilot as a provider. However, I'm unsure if using the API would provide the same level of "quality" as Copilot.
